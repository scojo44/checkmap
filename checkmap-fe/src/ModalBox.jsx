import React from "react"
import {Outlet, useLocation} from "react-router-dom";
import Modal from 'react-modal'

/** Wrapper to show forms as a modal dialog box */

export default function ModalBox({element}) {
  const location = useLocation();
  const previousLocation = location.state?.previousLocation; // Used to tell ModalBox when to be open
  
  // Bind modal to your appElement (https://reactcommunity.org/react-modal/accessibility/)
  Modal.setAppElement('#root');

  const customStyles = {
    overlay: {
      backgroundColor: 'rgba(0,0,0,50%)',
      zIndex: '9999'
    },
    content: {
      top: '5%',
      bottom: 'auto',
      width: '50%',
      margin: 'auto',
      minWidth: '400px',
      borderRadius: '1rem'
    },
  };

  return (
    <Modal
      isOpen={!!previousLocation}
      onAfterOpen={afterOpenModal}
      onRequestClose={closeModal}
      style={customStyles}
      contentLabel="Example Modal"
    >
      <Outlet/>
    </Modal>
  );

  function closeModal() {
    // Save in case it's needed
  }

  function afterOpenModal() {
    // references are now sync'd and can be accessed.
    // subtitle.style.color = '#f00';
  }
}
