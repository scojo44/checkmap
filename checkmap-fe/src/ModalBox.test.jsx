import {render} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import ModalBox from './ModalBox'
import Modal from 'react-modal';


describe('ModalBox Tests', () => {
  // React-Modal expects a top-level element with ID of 'root' for accessibility
  // Adding a <div id="root"> around the <ModalBox> doesn't work.
  Modal.setAppElement = vi.fn();

  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <ModalBox element={<h1>Test</h1>}/>
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
          <ModalBox element={<h1>Test</h1>}/>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
