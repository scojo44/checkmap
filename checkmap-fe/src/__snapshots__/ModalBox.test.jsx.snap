// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ModalBox Tests > Matches snapshot 1`] = `<DocumentFragment />`;
